pkg/api/v1/types.go

```diff
@@ -2026,16 +2026,16 @@ type NodeSelector struct {
 // A null or empty node selector term matches no objects.
 type NodeSelectorTerm struct {
 	//Required. A list of node selector requirements. The requirements are ANDed.
-	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" protobuf:"bytes,1,rep,name=matchExpressions"`
+	// +patchMergeKey=key
+	// +patchStrategy=merge
+	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" patchStrategy:"merge" patchMergeKey:"key" protobuf:"bytes,1,rep,name=matchExpressions"`
```

Comment:
@liggitt I believe you're correct (didn't check when this code went in, but it's definitely shipped). And I agree that the behavior with skewed versions would be as you describe.

On the other hand, clients expectations (based on the misplaced annotations) is that it should probably perform merges. So I suppose the question is whether we would be willing to tolerate *different* (and admittedly slightly worse) broken behavior when using skewed clients/servers in exchange for fixing the behavior to match expectations.

Otherwise I think this is going to be broken forever and the correct fix is to just remove the misplaced annotations so that it doesn't confuse people.

cc @aaronlevy